File: application-5/senaryo-c#/dbfactory.cs
using System;

namespace DatabaseFactory
{
    public class DBFactory
    {
        public const int MYSQL = 1;
        public const int ORACLE = 2;
        public const int MSSQL = 3;

        public DB GetDB(int type)
        {
            switch (type)
            {
                case MYSQL:
                    return new MYSQL();
                case ORACLE:
                    return new ORACLE();
                case MSSQL:
                    return new MSSQL();
                default:
                    return null;
            }
        }
    }
}

File: application-5/senaryo-pyhton/db_factory.py
from mysql import MYSQL
from oracle import ORACLE
from mssql import MSSQL

class DBFactory:
    MYSQL = 1
    ORACLE = 2
    MSSQL = 3

    @staticmethod
    def get_db(db_type):
        if db_type == DBFactory.MYSQL:
            return MYSQL()
        elif db_type == DBFactory.ORACLE:
            return ORACLE()
        elif db_type == DBFactory.MSSQL:
            return MSSQL()
        else:
            return None
